diff --git a/src/daemon.c b/src/daemon.c
index 9e59b83..c4878fb 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -1803,10 +1803,11 @@ daemon_create_user_backend (Daemon         *daemon,
         g_autofree char *admin_groups = NULL;
         GError *inner_error = NULL;
         User *user = NULL;
-        const gchar *argv[9] = { NULL };
+        const gchar *argv[10] = { NULL };
         int i = 0;
 
         argv[i++] = "/usr/sbin/useradd";
+        argv[i++] = "--extrausers";
         argv[i++] = "-m";
         argv[i++] = "-c";
         argv[i++] = cd->real_name;
@@ -2026,7 +2027,7 @@ daemon_delete_shadow_user (Daemon                *daemon,
         const gchar *username;
         const gchar *homedir;
         gchar *resolved_homedir;
-        const gchar *argv[6];
+        const gchar *argv[7];
         int i = 0;
 
         username = accounts_user_get_user_name (ACCOUNTS_USER (user));
@@ -2041,6 +2042,7 @@ daemon_delete_shadow_user (Daemon                *daemon,
         free (resolved_homedir);
 
         argv[i++] = "/usr/sbin/userdel";
+        argv[i++] = "--extrausers";
         argv[i++] = "-f";
 
         if (rm_files)
diff --git a/src/meson.build b/src/meson.build
index 2a09584..3681f96 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -32,8 +32,8 @@ cflags = [
   '-DLOCALSTATEDIR="@0@"'.format(act_localstatedir),
   '-DDATADIR="@0@"'.format(act_datadir),
   '-DSYSCONFDIR="@0@"'.format(act_sysconfdir),
-  '-DICONDIR="@0@"'.format(join_paths(act_localstatedir, 'lib', 'AccountsService', 'icons')),
-  '-DUSERDIR="@0@"'.format(join_paths(act_localstatedir, 'lib', 'AccountsService', 'users')),
+  '-DICONDIR="@0@"'.format(join_paths('/etc', 'writable', 'AccountsService', 'icons')),
+  '-DUSERDIR="@0@"'.format(join_paths('/etc', 'writable', 'AccountsService', 'users')),
 ]
 
 libaccounts_shared = static_library(
diff --git a/src/util.c b/src/util.c
index f99dfac..2645493 100644
--- a/src/util.c
+++ b/src/util.c
@@ -422,7 +422,7 @@ init_dirs (void)
 
         userdir = g_strdup (USERDIR);
         icondir = g_strdup (ICONDIR);
-        sysconfdir = g_strdup ("/etc/");
+        sysconfdir = g_strdup ("/var/lib/extrausers");
 }
 
 void
